bootloader/assets/assets.go

```diff
@@ -41,6 +51,51 @@ func Internal(name string) []byte {
 	return registeredAssets[name]
 }
 
+type byFirstEdition []forEditions
+
+func (b byFirstEdition) Len() int           { return len(b) }
+func (b byFirstEdition) Swap(i, j int)      { b[i], b[j] = b[j], b[i] }
+func (b byFirstEdition) Less(i, j int) bool { return b[i].FirstEdition < b[j].FirstEdition }
+
+// registerSnippetForEditions register a set of snippets, each carrying the
+// first edition number it applies to, under a given key.
+func registerSnippetForEditions(name string, snippets []forEditions) {
```

Comment:
it might be simpler/clearer to check that snippets is already sorted and panic if not, instead of sorting them. It will be confusing for the reader if they don't appear sorted